### PR TITLE
Feat: Active Group

### PR DESCRIPTION
**Pull request by winged**
Add a new `activeGroup` filter to tell the backend who the user is currently acting as.
Also allow the frontend to pass in `modifiedByGroup` and `createdByGroup` parameters, and validate those against the user's actual groups

### CODE REVIEW

File: alexandria/core/tests/test_views.py [hunk 1]
```diff
@@ -196,3 +198,44 @@ def test_hook_view(
...
+    post_data = {
+        # Note: model name pluralisation may not always be with an "s" suffix,
+        # but for now, this holds true
+        "data": {"attributes": serialized_model, "type": f"{model_name}s"}
+    }
+
+    url = reverse(f"{model_name}-list")
+
+    if expect_failure:
+        # with pytest.raises(ValidationError):
```
Comment (open-dynaMIX) [inline alexandria/core/tests/test_views.py:234]: Forgot to cleanup?

**Review by open-dynaMIX — CHANGES_REQUESTED** [review]
Nice work! :tada:

File: alexandria/core/tests/test_filters.py [hunk 2]
```diff
@@ -95,3 +97,38 @@ def test_tag_category_filter(db, document_factory, tag_factory, admin_client):
     result = resp.json()
     assert len(result["data"]) == 1
     assert result["data"][0]["id"] == str(blue.pk)
+
+
+@pytest.mark.parametrize(
+    "admin_groups, active_group, expect_failure",
+    [
+        (["foo", "bar"], "foo", False),
```
[the remark this answers is not in the document]
Comment (winged) [reply alexandria/core/tests/test_filters.py:105]: This is a fixture used by `admin_user`, which in turn is used by `admin_client` :D

File: alexandria/core/tests/test_views.py (hunk 1 above)
Comment (winged) [reply alexandria/core/tests/test_views.py]: Yep, fixed

File: alexandria/core/tests/test_filters.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (open-dynaMIX) [reply alexandria/core/tests/test_filters.py:105]: Ah, now I understand.

**Review by open-dynaMIX — APPROVED** [review]

**Pull request merged**